balancer-js/src/modules/sor/pool-data/subgraphPoolDataService.ts

```diff
@@ -92,3 +103,61 @@ export class SubgraphPoolDataService implements PoolDataService {
     return pools;
   }
 }
+
+export class OnChainPoolsRepository implements PoolDataService {
```

Comment:
This is basically the same as `SubgraphPoolDataService` above. I think there should be a way to reuse?

Reply:
Looking again at the code, I feel like I should be using a repository instead of a data service directly. If I'm not mistaken, I should be able to use the `PoolsSubgraphOnChainRepository` that you implemented. Please let me know if you think this is not the case.

Reply:
I decided to simply remove `OnChainPoolsRepository` and slightly adjust `SubgraphPoolDataService` to use it as is. I think we should eventually consolidate this data service with `PoolsSubgraphOnChainRepository`, but I believe that can be done on another PR, so we don't delay this one even further.